pkg/capture/capture_manager.go

```diff
@@ -211,61 +252,92 @@ func (cm *Manager) Update(ctx context.Context, ifaces config.Ifaces) {
+	var enableIfaces = append(enable, update...)
+
+	// execute a final writeout of all disabled interfaces in the list
+	if len(disableIfaces) > 0 {
+		cm.performWriteout(ctx, time.Now().Add(time.Second), disableIfaces...)
+	}
+
 	// To avoid any interference the update() logic is protected as a whole
 	// This also allows us to interace with the captures without copying (creating potential races)
 	cm.Lock()
 	defer cm.Unlock()
 
-	// Enable any interfaces present in the positive list
-	var rg RunGroup
-	for _, iface := range enable {
+	// store the configuration so that changes can be communicated
+	cm.lastAppliedConfig = ifaces
+
+	// Disable any interfaces present in the negative list
+	var disableRg RunGroup
+	for _, iface := range disableIfaces {
 		iface := iface
-		rg.Run(func() {
 
-			runCtx := logging.WithFields(ctx, slog.String("iface", iface))
+		mc, exists := cm.captures.Get(iface)
+		if !exists {
+			continue
+		}
+		disableRg.Run(func() {
+
+			runCtx := logging.WithFields(ctx, "iface", mc.iface)
 
-			cap := newCapture(iface, ifaces[iface]).SetSourceInitFn(cm.sourceInitFn)
-			if err := cap.run(runCtx); err != nil {
-				logging.FromContext(runCtx).Errorf("failed to start capture: %s", err)
+			logger := logging.FromContext(runCtx)
+			logger.Info("closing capture / stopping packet processing")
+
+			if err := mc.close(); err != nil {
+				logger.Errorf("failed to close capture: %s", err)
 				return
 			}
 
-			cm.captures.Set(iface, cap)
+			cm.captures.Delete(mc.iface)
 		})
 	}
-	rg.Wait()
+	disableRg.Wait()
+
+	// NOTE: make sure to use a different run group than disable's. Not doing it
+	// lead to captures being enabled while they were closed, leading to
+	//
+	// "failed to free capture resources: cannot call Free() on open capture source, call Close() first"
+	// errors
+	var enableRg RunGroup
+	// Enable any interfaces present in the positive list
+	for _, iface := range enableIfaces {
+		iface := iface
 
-	// Disable any interfaces present in the negative list
-	for _, iface := range disable {
-		mc, exists := cm.captures.Get(iface)
-		if !exists {
-			continue
-		}
-		rg.Run(func() {
+		enableRg.Run(func() {
 
-			runCtx := logging.WithFields(ctx, slog.String("iface", mc.iface))
+			runCtx := logging.WithFields(ctx, "iface", iface)
+			logger := logging.FromContext(runCtx)
 
-			if err := mc.close(); err != nil {
-				logging.FromContext(runCtx).Errorf("failed to close capture: %s", err)
+			logger.Info("initializing capture / running packet processing")
+
+			cap := newCapture(iface, ifaces[iface]).SetSourceInitFn(cm.sourceInitFn)
+			if err := cap.run(runCtx); err != nil {
+				logger.Errorf("failed to start capture: %s", err)
 				return
 			}
 
-			cm.captures.Delete(mc.iface)
+			cm.captures.Set(iface, cap)
 		})
 	}
-	rg.Wait()
+	enableRg.Wait()
 
 	logger.With(
 		"elapsed", time.Since(t0).Round(time.Millisecond).String(),
-		"ifaces_added", enable,
-		"ifaces_removed", disable,
+		slog.Group("ifaces",
```

Comment:
It's the way that the logging context is constructed. Right now, it still relies on the `key, value, key, value, ...` list. That's why it didn't pick up on just `slog.String(key, value)`. I wonder if we should change the function signature to not accept anything, but just `...slog.Attr`. What do you think?

Reply:
IMHO it would improve clarity / readability. Something implicit like `key, value, key, value` doesn't. But I don't have strong feelings. Regardless, ideally we stick to the same concept everywhere (whatever we choose), be it by _forcing_ it or not...